### PR TITLE
Testing improvements: use llvm-cov and rstest

### PR DESCRIPTION
**Pull request by peteretelej**
Testing infrastructure improvements. This change updates testing to use `cargo-llvm-cov` which has better coverage reporting, and easier to work with across platforms. Also adds rstest for use in parameterized tests and using it to cover some previously uncovered features.

## Summary by CodeRabbit

* **Tests**
  * Expanded comprehensive test coverage for CLI argument handling, file parsing, display colorization, and directory traversal; added many parameterized and edge-case tests (sorting, filtering, formatting, symlinks, executables).

* **Refactor**
  * Promoted parsing utilities for 7z, RAR, and path normalization to public use to enable broader reuse and testing.

* **Chores**
  * Switched CI coverage tooling to an LLVM-based reporter and added a test helper dependency for parameterized tests.

### CODE REVIEW

File: tests/fromfile_unit_tests.rs [hunk 1]
```diff
@@ -176,3 +178,103 @@ fn test_parse_zip_listing() {
...
+    assert_eq!(entry.size, Some(9));
+}
+
+#[test]
+fn test_parse_rar_line_verbose_directory() {
+    let line = " drwxrwxr-x         0         0   0%  2025-07-26 21:18  00000000  test_rar/";
+    let entry = parse_rar_line(line).unwrap();
+    assert_eq!(entry.path, "test_rar/");
+    assert!(entry.is_dir);
+}
```
[the remark this answers is not in the document]
Comment (peteretelej) [reply tests/fromfile_unit_tests.rs]: wow, good bot
